File: apps/web/src/types/interview-pack.ts
export interface InterviewQuestion {
  id: string;
  jdRequirement: string; // The job description requirement this question targets
  candidateEvidence: string | null; // Snippet from CV that relates, or null if "missing"
  mainQuestion: string;
  followUpQuestions: string[];
  goodLooksLike: string; // What a good answer looks like
  redFlags: string[]; // Red flags to watch for
  timebox: number; // Minutes allocated
  rubric: {
    [key: string]: string; // Score 1-5 with descriptions
  };
  notes?: string; // Interviewer's notes for this question
  score?: number; // Score 1-5 for this question
}

export interface InterviewPack {
  id: string;
  duration: 30 | 60 | 90; // Interview duration in minutes
  focusAreas: string[]; // Key areas to focus on
  risks: string[]; // Risks/concerns identified
  timeline: string; // Suggested interview timeline breakdown
  questions: InterviewQuestion[];
  createdAt: string;
  updatedAt: string;
}

export interface GenerateInterviewPackRequest {
  jobDescription: string;
  jobTitle: string;
  candidateData: {
    name: string;
    experience: any[];
    education: any[];
    skills: any[];
    certifications: any[];
    cvText?: string;
  };
  duration: 30 | 60 | 90;
}

File: apps/api/app/models/interview_pack.py
"""Interview pack models."""
from typing import List, Dict, Optional
from pydantic import BaseModel, Field


class InterviewQuestion(BaseModel):
    """A single interview question."""
    jd_requirement: str = Field(..., description="The job description requirement this question targets")
    candidate_evidence: Optional[str] = Field(None, description="Snippet from CV that relates, or null if missing")
    main_question: str = Field(..., description="Main interview question")
    follow_up_questions: List[str] = Field(default_factory=list, description="Follow-up questions")
    good_looks_like: str = Field(..., description="What a good answer looks like")
    red_flags: List[str] = Field(default_factory=list, description="Red flags to watch for")
    timebox: int = Field(..., ge=5, le=30, description="Minutes allocated for this question")
    rubric: Dict[str, str] = Field(..., description="Score 1-5 with descriptions")


class InterviewPackRequest(BaseModel):
    """Request to generate an interview pack."""
    job_description: str = Field(..., description="Job description text")
    job_title: str = Field(..., description="Job title")
    candidate: dict = Field(..., description="Candidate data (ParsedCandidate format)")
    duration: int = Field(..., ge=30, le=90, description="Interview duration in minutes (30, 60, or 90)")
    interview_type: Optional[str] = Field(None, description="Type of interview (technical, behavioral, phone_screen, final, other)")


class InterviewPackResponse(BaseModel):
    """Response containing generated interview pack."""
    request_id: str
    pack: dict = Field(..., description="Interview pack data")
    processing_time_ms: float

